mycore-neo4j/src/main/java/org/mycore/mcr/neo4j/datamodel/metadata/neo4jutil/Neo4JRelation.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * This file is part of ***  M y C o R e  ***
+ * See http://www.mycore.de/ for details.
+ *
+ * MyCoRe is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MyCoRe is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MyCoRe.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.mycore.mcr.neo4j.datamodel.metadata.neo4jutil;
+
+import static org.mycore.mcr.neo4j.datamodel.metadata.neo4jutil.MCRNeo4JUtil.removeIllegalRelationshipTypeCharacters;
+
+import java.util.concurrent.ThreadLocalRandom;
+
+public record Neo4JRelation(String sourceNodeID, String targetNodeID, String relationshipType) {
+
+      @Override
+      public String toString() {
+            return "Neo4JRelation{" +
+                  "sourceNode='" + sourceNodeID + '\'' +
+                  ", targetNode='" + targetNodeID + '\'' +
+                  ", relationshipType='" + relationshipType + '\'' +
+                  '}';
+      }
+
+      public String toAppendQuery() {
+            StringBuilder sbQuery = new StringBuilder();
+            Long random = ThreadLocalRandom.current().nextLong(Long.MAX_VALUE);
+            sbQuery.append("MERGE (b").append(random).append(" {id: '").append(targetNodeID).append("'})\n")
```

Comment:
Maybe the queries become more readable with text blocks (https://www.baeldung.com/java-multiline-string#text-blocks) and use of String.format() with position arguments:
```
// Specifying argument positions. %1$ is for the first argument and %2$ is for the second argument */
String fstr = String.format("My String is: %1$s, %1$s and %2$s", "cool string", "88");
//fstr = "My String is: cool string, cool string and 88"
```
}

Output:

Reply:
Hab eine Java Doc Beschreibung zur besseren Erklärung hinzugefügt, da sich schon weiter unten gegen die Verwendung der Cypher-DSL ausgesprochen wurde.